Library/Homebrew/cmd/update-bash.sh

```diff
@@ -320,8 +333,23 @@ EOS
   # this procedure will be removed in the future if it seems unnecessary
   rename_taps_dir_if_necessary
 
-  # kill all of subprocess on interrupt
-  trap '{ pkill -P $$; wait; exit 130; }' SIGINT
+  # Spawn ssh-agent if it's not running and at least one of repos is using ssh
+  # https://github.com/Homebrew/homebrew/issues/48267
+  for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
+  do
+    [[ -d "$DIR/.git" ]] || continue
+    cd "$DIR" || continue
+    [[ "$(git config --get remote.origin.url 2>/dev/null)" =~ ^https?:// ]] && continue
```

Comment:
I don't believe so but it's definitely not for SSH. There may be a way of querying Git directly for which protocol it'll use?